music_assistant/providers/plex/__init__.py

```diff
@@ -1001,3 +1048,55 @@ def _refresh_plex_token() -> MyPlexAccount:
             return self._myplex_account
 
         return await asyncio.to_thread(_refresh_plex_token)
+
+    async def set_favorite(self, prov_item_id: str, media_type: MediaType, favorite: bool) -> None:
+        """Set favorite status by setting rating in Plex."""
+        if favorite:
+            # Set like rating
+            rating = cast("float", self.config.get_value(CONF_PLEX_LIKE_RATING))
+        else:
+            # Set unlike rating
+            rating = cast("float", self.config.get_value(CONF_PLEX_UNLIKE_RATING))
+
+        if media_type == MediaType.TRACK:
+            plex_track = await self._get_data(prov_item_id, PlexTrack)
+            await self._run_async(plex_track.rate, rating)
+            self.logger.debug(
+                "Set Plex rating to %s for track with ID %s (ratingKey: %s)",
+                rating,
+                prov_item_id,
+                plex_track.ratingKey,
+            )
+        elif media_type == MediaType.ALBUM:
+            plex_album = await self._get_data(prov_item_id, PlexAlbum)
+            await self._run_async(plex_album.rate, rating)
+            self.logger.debug(
+                "Set Plex rating to %s for album with ID %s (ratingKey: %s)",
+                rating,
+                prov_item_id,
+                plex_album.ratingKey,
+            )
+
+    async def library_add(self, item: MediaItemType) -> bool:
+        """Add item to library by setting the configured like rating in Plex."""
+        if item.media_type in (MediaType.TRACK, MediaType.ALBUM):
+            await self.set_favorite(item.item_id, item.media_type, True)
+            return True
+        self.logger.info(
+            "Provider %s does not support library_add for media type %s",
+            self.name,
+            item.media_type,
+        )
+        return False
+
```

Comment:
```suggestion
```

see my comment on library_remove - you don't need to implement this, you only need the set_favorite method